src/solver/conv_mlir_igemm_fwd_xdlops.cpp

```diff
@@ -139,14 +151,35 @@ bool PerformanceConvMlirIgemmXdlops::SetNextValue(const ConvolutionContext& /*co
             break;
         if(!NextTwoPower<16, 256>(GemmNPerBlock))
             break;
-        if(!NextTwoPower<1, 8>(GemmKPerBlock))
+        if(!NextTwoPower<4, 8>(GemmKPerBlock))
             break;
         if(!NextTwoPower<4, 128>(GemmMPerWave))
             break;
         if(!NextTwoPower<16, 128>(GemmNPerWave))
             break;
-        if(!NextTwoPower<4, 8>(GemmKPACKSize))
-            break;
+        if(config.IsFp32())
+        {
+            if(config.direction.IsForward())
+            {
+                if(!NextTwoPower<1, 4>(GemmKPACKSize))
+                    break;
+            }
+            else if(config.direction.IsBackwardWrW())
+            {
+                if(!NextTwoPower<1, 2>(GemmKPACKSize))
+                    break;
+            }
+            else if(config.direction.IsBackwardData())
+            {
+                if(!NextTwoPower<1, 1>(GemmKPACKSize))
+                    break;
+            }
+        }
+        else if(config.IsFp16())
+        {
+            if(!Next_1_4_8(GemmKPACKSize))
```

Comment:
Could you add a comment here to link to the MLIR ticket and mention that this aim to be reverted?

Reply:
This does not aim to be reverted. I assume you are talking about fp32 wrw?

Reply:
>  I assume you are talking about fp32 wrw?

Right, I got myself confused. Mainly because we skipped 2.

I think it will help explain a lot if you can put the PR description on why not include 2 here.

Reply:
Revised the description on why KPACK=2 is not used for fp16.